homeworks/lesia.turua_lesiaturuta/PokemonProject/scripts/pokemon.js

```diff
@@ -309,58 +66,138 @@ const renderCards = (cardContainer, pokemonsArray) => {
     return true;
 };
 
-const cleanElement = (el) => {
-    while (el.firstChild) {
-        el.removeChild(el.firstChild);
+const renderSelect = (name) => {
+    cleanElement(searchEl);
+    const optionFirstEl = document.createElement('option');
+    if (name === '') {
+        optionFirstEl.value = 'Select type: ';
+        optionFirstEl.text = 'Select type: ';
+        optionFirstEl.selected = true;
+        optionFirstEl.disabled = true;
+    } else {
+        optionFirstEl.value = '';
+        optionFirstEl.text = 'All types';
+    }
+
+    searchEl.add(optionFirstEl);
+    typesPokemons.forEach((type) => {
+        const optionEl = document.createElement('option');
+        optionEl.value = type;
+        if (type === name) {
+            optionEl.selected = true;
+        }
+        optionEl.text = type;
+        searchEl.add(optionEl);
+    });
+};
+
+const sortPokemonsUp = (arrayPokemons) => arrayPokemons.sort((a, b) => a.height - b.height);
+
+const sortPokemonsDown = (arrayPokemons) => arrayPokemons.sort((a, b) => b.height - a.height);
+
+const updateTypes = (array) => {
+    const types = [];
+    array.forEach((pokemon) => pokemon.type.forEach((type) => {
+        if (!types.includes(type)) types.push(type);
+    }));
+    return types;
+};
+
+const getPokemonsByType = (type, arrayPokemons) => arrayPokemons
+    .filter((pokemon) => pokemon.type
+        .includes(type));
+const getPokemonsByValet = (text, arrayPokemons) => arrayPokemons
```

Comment:
```suggestion
const getPokemonsBySearchName = (text, arrayPokemons) => arrayPokemons
```
`Valet` - what does it mean?

Reply:
you can add a logic of empty search string here
```
if (text === '') return arrayPokemon
```